### PR TITLE
Fix bug allowing votes on archived content

### PR DESCRIPTION
**Pull request by DogPawHat**
This is a fix for issue #176. There is an archived flag in the JSON for posts and comments, so all this pr does is add support for the archived flag and place a few if statements to block votes and show error messages in toasts.

This is my first ever PR and while I'm not necessarily a beginner programmer, my android dev experience is limited, so let me know if this has to be changed around.

### CODE REVIEW

File: src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java [hunk 1]
```diff
@@ -712,6 +730,16 @@ public void run() {
 
 		refreshView(activity);
 
+		boolean vote = (action == RedditAPI.RedditAction.DOWNVOTE
+				| action == RedditAPI.RedditAction.UPVOTE
+				| action == RedditAPI.RedditAction.UNVOTE);
+
+		if(src.archived && vote){
+			Toast.makeText(activity, "sorry, this has been archived and can no longer be voted on", Toast.LENGTH_SHORT)
```
Comment (QuantumBadger) [inline src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java]: Strings should be added to the end of strings.xml:

https://github.com/QuantumBadger/RedReader/blob/master/res/values/strings.xml

File: src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedComment.java [hunk 2]
```diff
@@ -272,15 +273,37 @@ public void action(final Activity activity, final RedditAPI.RedditAction action)
...
 		}
 
 		refreshView(activity);
 
+		boolean vote = (action == RedditAPI.RedditAction.DOWNVOTE
+				| action == RedditAPI.RedditAction.UPVOTE
+				| action == RedditAPI.RedditAction.UNVOTE);
+
+		if(src.archived && vote){
+			Toast.makeText(activity, "sorry, this has been archived and can no longer be voted on", Toast.LENGTH_SHORT)
```
Comment (QuantumBadger) [inline src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedComment.java]: See other comment re: strings

File: src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java [hunk 3]
```diff
@@ -694,9 +694,27 @@ public void run() {
...
 
 		switch(action) {
-			case DOWNVOTE: voteDirection = -1; break;
-			case UNVOTE: voteDirection = 0; break;
-			case UPVOTE: voteDirection = 1; break;
+			case DOWNVOTE:
+				if(!src.archived) {
+					voteDirection = -1;
+				} else {
+					voteDirection = lastVoteDirection;
```
Comment (QuantumBadger) [inline src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java]: I think these else clauses can be removed -- {{voteDirection}} doesn't need to be set at all here.

**Comment by QuantumBadger** [comment]
Looks good, thanks! I've added a couple of minor comments.

**Comment by DogPawHat** [comment]
Should be ok now.

**Comment by QuantumBadger** [comment]
Thanks!

**Pull request merged**